### ISSUE TITLE
LineItem.java #684  hasDiscounts() Returns Null

### ISSUE THREAD

**Issue by bryannevega** [clover/clover-android-sdk#80]
If the object doesn't have any discounts the line item object doesn't have a "hasDiscounts" boolean field nor the method returns false.

In order to avoid the issue, I use 
`lineItem.getDiscounts != null `
where it could be :
```
if(lineItem.hasDiscounts){
//Logic
}
```

Is this the intended behavior or can this be considered into implementation?

**Comment by AchinthaBandara** [comment]
To use in the if condition, it should return a Boolean value, in your case it does not.

**Comment by bryannevega** [comment]
> 
> 
> To use in the if condition, it should return a Boolean value, in your case it does not.

No, it always returns a NPE because the object itself doesn't have a field "hasDiscounts".
